app/src/main/java/org/mozilla/fenix/library/bookmarks/edit/EditBookmarkFragment.kt

```diff
@@ -147,19 +149,35 @@ class EditBookmarkFragment : Fragment(), CoroutineScope {
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
         return when (item.itemId) {
             R.id.delete_bookmark_button -> {
-                launch(IO) {
-                    requireComponents.core.bookmarksStorage.deleteNode(guidToEdit)
-                    requireComponents.analytics.metrics.track(Event.RemoveBookmark)
-                    launch(Main) {
-                        Navigation.findNavController(requireActivity(), R.id.container).popBackStack()
-                    }
-                }
+                displayDeleteBookmarkDialog()
                 true
             }
             else -> super.onOptionsItemSelected(item)
         }
     }
 
+    private fun displayDeleteBookmarkDialog() {
+        activity?.let { activity ->
+            AlertDialog.Builder(activity).apply {
+                setMessage(R.string.bookmark_deletion_confirmation)
+                setNegativeButton(android.R.string.cancel) { dialog: DialogInterface, _ ->
+                    dialog.cancel()
+                }
+                setPositiveButton(R.string.tab_collection_dialog_positive) { dialog: DialogInterface, _ ->
+                    launch(IO) {
+                        requireComponents.core.bookmarksStorage.deleteNode(guidToEdit)
+                        requireComponents.analytics.metrics.track(Event.RemoveBookmark)
+                        launch(Main) {
+                            Navigation.findNavController(requireActivity(), R.id.container).popBackStack()
```

Comment:
Nit: We should use the far simpler Fragment.findNavController() API that they added to the Navigation library. https://developer.android.com/reference/kotlin/androidx/navigation/fragment/package-summary#(androidx.fragment.app.Fragment).findNavController()